src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+            (player->actor.bgCheckFlags & 1)) {
+            sp1C = 1;
+        } else {
+            sp1C = 2;
+        }
+    } else {
+        return 0;
+    }
+
+    triggerEntry = &D_801BFFA0[sp1C];
+    if (!Player_InCsMode(globalCtx)) {
+        if ((triggerEntry->flag) && !(gSaveContext.textTriggerFlags & triggerEntry->flag) && (sp1C == 0)) {
+            func_801518B0(globalCtx, triggerEntry->textId, NULL);
+            gSaveContext.textTriggerFlags |= triggerEntry->flag;
+        }
+    }
+
+    return sp1C + 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123DA4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123DC0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123E90.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123F14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123F2C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123F48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124020.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124420.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/Player_GetMask.s")
+void func_80124618(struct_80124618 arg0[], f32 curFrame, Vec3f* arg2) {
+    s32 currentFrame = curFrame;
+    f32 temp_f0;
+    f32 temp_f14;
+    f32 progress;
+    s16 temp_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/Player_RemoveMask.s")
+    do {
+        temp_v1 = arg0[1].unk_0;
+        arg0++;
+    } while (temp_v1 < currentFrame);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_8012405C.s")
+    temp_f0 = arg0[-1].unk_0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124088.s")
+    progress = (curFrame - temp_f0) / (((f32)temp_v1) - temp_f0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801240C8.s")
+    temp_f14 = arg0[-1].unk_2.x;
+    arg2->x = LERPIMP(temp_f14, arg0->unk_2.x, progress) * 0.01f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801240DC.s")
+    temp_f14 = arg0[-1].unk_2.y;
+    arg2->y = LERPIMP(temp_f14, arg0->unk_2.y, progress) * 0.01f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124110.s")
+    temp_f14 = arg0[-1].unk_2.z;
+    arg2->z = LERPIMP(temp_f14, arg0->unk_2.z, progress) * 0.01f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124148.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801246F4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124168.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124870.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124190.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80124CC4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801241B4.s")
+extern s32 D_801C0958;
+#if 0
+s32 D_801C0958 = false;
+#endif
```

Comment:
Is there a reason this is here rather than further down? I've noticed a few bits of data in odd places in here.

Reply:
I reused the placement of data in the main player_lib branch